### PR TITLE
Fix leaking goroutines in multiple integration tests 

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

Note to reviewer: I recommend reviewing commits one by one - they are all independent.

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig storage

### CODE REVIEW

File: pkg/controller/volume/attachdetach/attach_detach_controller.go [hunk 1]
```diff
@@ -913,7 +916,7 @@ func (adc *attachDetachController) GetNodeName() types.NodeName {
 }
 
 func (adc *attachDetachController) GetEventRecorder() record.EventRecorder {
-	return adc.recorder
+	return nil
```
Comment (aojea) [inline pkg/controller/volume/attachdetach/attach_detach_controller.go:919]: this seems to break some assumptions

**Comment by aojea** [comment]
/retest

lgtm , just one question about what happens with the code that was assuming they will receive an event recorder and now receive nil https://github.com/kubernetes/kubernetes/pull/110633#discussion_r899892525

File: pkg/controller/volume/attachdetach/attach_detach_controller.go (hunk 1 above)
Comment (wojtek-t) [reply pkg/controller/volume/attachdetach/attach_detach_controller.go:916]: This was never set anywhere - so it's supposed to be a no-op change :)

[As you see - I removed `recorder` field completely and didn't have to change any code for it.]

**Comment by wojtek-t** [comment]
/retest

@aojea - this is now ready for your review

File: pkg/controller/volume/attachdetach/attach_detach_controller.go (hunk 1 above)
Comment (aojea) [reply pkg/controller/volume/attachdetach/attach_detach_controller.go:916]: yeah, it's is weird, is just to meet some Interface or something?, the `GetNodeName()` above is returning empty too

```
grep -r GetEventRecorder pkg/controller/
pkg/controller/volume/attachdetach/attach_detach_controller.go:func (adc *attachDetachController) GetEventRecorder() record.EventRecorder {
pkg/controller/volume/expand/expand_controller.go:func (expc *expandController) GetEventRecorder() record.EventRecorder {
pkg/controller/volume/persistentvolume/volume_host.go:func (ctrl *PersistentVolumeController) GetEventRecorder() record.EventRecorder {
```

**Comment by aojea** [comment]
> W0718 11:06:41.740] scp: /var/log/startupscript.log*: No such file or directory
> W0718 11:06:41.742] ERROR: (gcloud.compute.scp) [/usr/bin/scp] exited with return code [1].
> W0718 11:06:42.809] scp: /var/log/fluentd.log*: No such file or directory
> W0718 11:06:42.888] scp: /var/log/kubelet.cov*: No such file or directory
> W0718 11:06:42.889] scp: /var/log/startupscript.log*: No such file or directory
> W0718 11:06:42.892] ERROR: (gcloud.compute.scp) [/usr/bin/scp] exited with return code [1].
> W0718 11:06:44.196] scp: /var/log/fluentd.log*: No such file or directory
> W0718 11:06:44.275] scp: /var/log/kubelet.cov*: No such file or directory
> W0718 11:06:44.275] scp: /var/log/startupscript.log*: No such file or directory
> W0718 11:06:44.278] ERROR: (gcloud.compute.scp) [/usr/bin/scp] exited with return code [1].
> W0718 11:06:48.322] INSTANCE_GROUPS=e2e-1c46229b00-a7d53-minion-group



/test pull-kubernetes-e2e-gce-ubuntu-containerd

**Comment by wojtek-t** [comment]
/retest

[tear down issue]

**Pull request merged**